examples/gadgets/snapshot/process/process.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/cilium/ebpf/rlimit"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/snapshot/process/tracer"
+)
+
+func main() {
+	if err := rlimit.RemoveMemlock(); err != nil {
+		return
+	}
+
+	processes, err := tracer.RunCollector(nil, nil)
```

Comment:
When passing all these `nil`, I cannot avoid thinking about how we can avoid them 😅. In this case, I immediately tought "How would it looks like using the _option pattern_?"

Definition:
```golang
func RunCollector(options ...Options) ([]processcollectortypes.Event, error) {
    ...
}
```

Usages:
```golang
processes, err := tracer.RunCollector(WithEnricher(t.helpers))
```
```golang
processes, err := tracer.RunCollector(WithEnricher(t.helpers), WithFiltering(mountNsMap))
```
```golang
processes, err := tracer.RunCollector()
```

It looks better, but I am not sure if it is worth using it in this case.

Reply:
I agree that it looks better, but it seems too much work now. How will it work for other categories? 

```
tracer, err := tracer.NewTracer(tracer.WithConfig(c), tracer.WithEnricher(foo))
```

?

Reply:
For other categories, the only parameter that is optional is the enricher so that it would be:

For `top` category:

```golang
tracer, err := tracer.NewTracer(config, statsCallback, errorCallback)
```
```golang
tracer, err := tracer.NewTracer(config, statsCallback, errorCallback, WithEnricher(t.helpers))
```

For `trace` category:
```golang
t.tracer, err = tracer.NewTracer(config, eventCallback)
```
```golang
t.tracer, err = tracer.NewTracer(config, eventCallback, WithEnricher(t.helpers))
```

Same for the `profile/cpu` gadget.

The implementation would be something like this:

```golang
type Option func(*Tracer)

func WithEnricher(enricher gadgets.DataEnricher) Option {
	return func(t *Tracer) {
		t.enricher = enricher
	}
}

func NewTracer(
	config *Config,
	eventCallback func(types.Event),
	options ...Option,
) (*Tracer, error) {
	t := &Tracer{
		config:        config,
		eventCallback: eventCallback,
	}

	for _, opts := range options {
		opts(t)
        }
	
        ...
}
```

However, given that the `Tracer` is different for each gadget, we would have to redefine `WithEnricher` within each gadget. Generics could solve this problem, but maybe it starts becoming too much just to avoid passing a `nil` 😅. It is more evident with the `RunCollector` where we pass three `nil`, but I don't have a better proposal for it now. Anyway, let's keep this in mind if we need to add more optional parameters to the tracers.